Task 5 added

Jhon112 opened the pull request:
add models/engine/file_storage.py, models/engine/__init__.py
Create FileStorage class with attributes and methods requested.
Edit models/init to instance FileStorage and have only one instance that gets and sets objs from file.json
I also edited to_dict method from BaseModel. 

> A copy was needed because we were not able to use storage.save()
> This was due to __dict__ being edited by adding it __class__ key,
> Also we were editing the value of updated_at and created_at keys
> so when we were going to dump the obj it has a datetime obj as
> value on updated_at key

monicajaimesc committed:
test

Jhon112 committed:
Task 4 completed. We use kwargs for constructor each key of this dictionary is an attribute name
each value of this dictionary is the value of this attribute name. created_at and updated_at are strings in kwargs,  convert these strings into datetime object.

Jhon112 committed:
merge task4

Jhon112 committed:
create init to convert folder in package

Jhon112 committed:
edit file to create a unique FileStorage instance

I can now import storage from models

Jhon112 committed:
main function to test task 5

Jhon112 committed:
Edit base_model class to use some storage methods

If kwargs are not passed public attr are set and storage.new is call.
This is done to create a new obj into our storage __objects attribute

From save method we call our storage.save method.
This writes the new instance to our JSON file that contains all objs

A change was made to our to_dict method as it was editing  __dict__.
This caused an issue when calling storage.new as it uses to_dict method.
More info on comment added to method

Jhon112 committed:
create init file to work engine as a package

Jhon112 committed:
Create FileStorage class.

Jhon112 committed:
Merge changes made to feature/BaseModel.
Fix import error and edit a bit the constructore structure

Jhon112 committed:
Merge branch 'master' into task-5

Jhon112 committed:
add shebang

monicajaimesc commented:
We have to check out this together, working in task 6 and fixing one bug in task 4. I won't accept the git merge as we talked ;)

Jhon112 committed:
import BaseModel on init file

This is useful to have the class available when importing models

Jhon112 committed:
Change import.

Import models package instead storage class.
This way when we will call storage only when needed and not before

Jhon112 committed:
Fix task5

the attributes are now class attributes therefore I removed init method

new method now stores the obj itself and not its dictionary

save method calls to_dict method of each obj to get a serializable str

reload method calls BaseModel with the kwargs gotten from json file

Jhon112 commented:
This version is now working as asked, it's creating the objects now and they are being printed as BaseModel objs and not as dictionaries, we'll review this code together to show what was wrong

monicajaimesc commented on models/engine/file_storage.py:
```diff
@@ -3,60 +3,60 @@
+	of the program. JSON files work as our database
+	"""
+
+	__file_path = 'file.json'
+	__objects = {}
+
```
Ok, cool.. I saw the same mistake

monicajaimesc reviewed (approved):
Cool, gonna try it in my consola and then we can talk about it. ;)

monicajaimesc committed:
Merge branch 'feature/BaseModel' into task-5
